contractcourt/contract_resolvers.go

```diff
@@ -530,8 +530,17 @@ func (h *htlcSuccessResolver) Resolve() (ContractResolver, error) {
 			return nil, fmt.Errorf("quitting")
 		}
 
+		// Finally, with the HTLC claimed, we'll attempt to settle its
+		// corresponding invoice if we were the original destination.
+		err = h.SettleInvoiceOnChain(h.payHash)
```

Comment:
Isn' t the original value in `h.htlcResolution.SweepSignDesc.Output.Value`  ?

Reply:
I looked into it, it for the local force case, it is the success tx amount, which is less than the htlc amount. For the remote case below, the number is the original htlc value.

But also for the local force case, the info doesn't need to be stored in the database. If I understand it correctly, `ChannelArbitrator.activeHTLCs` contains them. The original HTLC object could possibly be passed into this `Resolve` function.

I think it is important to not take the shortcut here by using the invoice amt.

Reply:
This data is also needed in https://github.com/lightningnetwork/lnd/pull/1875.

See commit `cnct: make original htlc amt available for contract resolvers`

Reply:
Why is the invoice amount a short cut? Isn't that what we want, the full value that was requested? The HTLC outputs for second level transactions are smaller than the _actual_ HTLC value since we pay fees from the HTLC value for that transaction. If the value after we subtract fees is below dust, then we consider that a dust output all together.

Reply:
> I went with just settling the invoice with whatever it originally requested

This is not the amount paid. It could be that more was paid.

Reply:
Ah yeah very true, over looked that!

Reply:
Final conclusion is that we can get this data from the logged `ChainActions`. They contain the original set of `channeldb.HTLC`s.

Reply:
For implementation, see commit above. Now using `ChainActions`.

Reply:
Sweet, thanks @joostjager! I believe this should be good to go now.